Closes #1622 - Value mismatches with numpy in operations test

Ethan-DeBandi99 opened the pull request:
Closes #1622 

Corrects the mismatch between numpy and arkouda values. This updates the return of division by zero to be `INFINITY` (in arkouda) which returns and treated as `inf` in numpy.

Output Before Updates
```
# ops implemented by arkouda but not numpy: 4
int64(array) >> uint64(array)  ->  [0 0 0 0 0 0 0 0 0 0]
int64(array) << uint64(array)  ->  [0 2 8 24 64 160 384 896 2048 4608]
uint64(array) >> int64(array)  ->  [0 0 0 0 0 0 0 0 0 0]
uint64(array) << int64(array)  ->  [0 2 8 24 64 160 384 896 2048 4608]
# ops implemented by both: 569
  Matching results:         564 / 569
  Arkouda execution errors: 0 / 569

  Dtype mismatches:         0 / 569

  Value mismatches:         5 / 569
int64(scalar) // float64(array): np: [inf 22. 11.  7.  5.  4.  3.  3.  2.  2.]
ak: [nan 22. 11.  7.  5.  4.  3.  3.  2.  2.]
uint64(scalar) // float64(array): np: [inf 22. 11.  7.  5.  4.  3.  3.  2.  2.]
ak: [nan 22. 11.  7.  5.  4.  3.  3.  2.  2.]
float64(scalar) // int64(array): np: [inf  3.  1.  1.  0.  0.  0.  0.  0.  0.]
ak: [nan  3.  1.  1.  0.  0.  0.  0.  0.  0.]
float64(scalar) // float64(array): np: [inf 14.  7.  4.  3.  2.  2.  2.  1.  1.]
ak: [nan 14.  7.  4.  3.  2.  2.  2.  1.  1.]
bool(scalar) // float64(array): np: [inf  4.  2.  1.  1.  0.  0.  0.  0.  0.]
ak: [nan  4.  2.  1.  1.  0.  0.  0.  0.  0.]
```

Output After Updates:
```
# ops implemented by arkouda but not numpy: 4
int64(array) >> uint64(array)  ->  [0 0 0 0 0 0 0 0 0 0]
int64(array) << uint64(array)  ->  [0 2 8 24 64 160 384 896 2048 4608]
uint64(array) >> int64(array)  ->  [0 0 0 0 0 0 0 0 0 0]
uint64(array) << int64(array)  ->  [0 2 8 24 64 160 384 896 2048 4608]
# ops implemented by both: 564
  Matching results:         564 / 564
  Arkouda execution errors: 0 / 564

  Dtype mismatches:         0 / 564

  Value mismatches:         0 / 564
```

pierce314159 reviewed (approved):
Looks great!

pierce314159 commented:
Ethan and I have talked offline, but we're gonna need more edge cases testing here. We never test the case where float scalar is `0.0`  or `np.inf` because the only scalars tested are the ones here
https://github.com/Bears-R-Us/arkouda/blob/1c32c3317c9afabb68967a418655a3c1c9b6fae9/tests/operator_tests.py#L31

Some examples of where we differ from numpy that aren't currently captured by our tests
```python
>>> a = ak.array([-np.inf,-7.0, 0.0, 7.0, np.inf])
>>> n = np.array([-np.inf,-7.0, 0.0, 7.0, np.inf])
>>> a//0.0
array([nan nan nan nan nan])

>>> n//0.0
array([-inf, -inf,  nan,  inf,  inf])

>>> 0.0//a
array([-0 -0 inf 0 0])

>>> 0.0//n
array([-0., -0., nan,  0.,  0.])

>>> a//np.inf
array([nan -0 0 0 nan])

>>> n//np.inf
 array([nan, -1.,  0.,  0., nan])

>>> np.inf//a
array([nan -inf inf inf nan])

>>> np.inf//n
array([nan, nan, inf, nan, nan])
```

Ethan-DeBandi99 commented:
Converting to draft until pierce finishes updated testing of edge cases and we configure the code to handle in accordance to numpy

Ethan-DeBandi99 commented:
Updated cases with new code that was merged after initially submitting PR. Added in tests @pierce314159 wrote. Everything is functional and matches the output of numpy.

pierce314159 reviewed (approved):
Looks great! Good work
